schema-engine/sql-schema-describer/src/postgres/indexes_query.sql

```diff
@@ -40,7 +39,7 @@ FROM
     INNER JOIN pg_class AS tableinfo ON tableinfo.oid = rawindex.indrelid
     INNER JOIN pg_class AS indexinfo ON indexinfo.oid = rawindex.indexrelid
     INNER JOIN pg_namespace AS schemainfo ON schemainfo.oid = tableinfo.relnamespace
-    INNER JOIN pg_attribute AS columninfo
+    LEFT JOIN pg_attribute AS columninfo
```

Comment:
This was a change from the initial pairing session I had with tom and unfortunately I don't remember 🥞
Reverting this change locally causes the following two tests to fail in that they are no longer detected as having expression indices 🤔
```
postgres::pg_xml_indexes_are_skipped,
gin::full_text_functions_filtered_out
```

Reply:
For a more in depth highlight of the failed tests, you can look at this [PR](https://github.com/prisma/prisma-engines/pull/4039)

Reply:
Meaning this is now not being detected any more (in case of `gin::full_text_functions_filtered_out`):
```
    let create_table = format!("CREATE TABLE \"{schema_name}\".\"A\" (id SERIAL PRIMARY KEY, data text not null)",);
    let create_idx =
        format!("CREATE INDEX \"A_data_idx\" ON \"{schema_name}\".\"A\" USING GIN (to_tsvector('english', data));",);
```

Next step would then be to manually run both queries (with LEFT and INNER joins) and compare the output.

Reply:
Okay, so @janpio and I did some further research.
Expression Indexes are generally defined without `column_info`. Due to this, we need to update the query to handle indexes where `column_info` is nullable, otherwise we lose expression indexes.

One thing I'd like to take note of in the highlight PR is that on updating `cargo t` to run with `--no-fail-fast`, we did also see the simple SQL test `postgres_index_with_expression_column` fail. A panic is raised as the logic that handles expression indexes was never triggered. This allowed indexes with `column_index != 0` through causing a panic. This is handled when we use the `LEFT JOIN` _in this case_ but still something to keep an eye on for future index work.